src/msg/async/frames_v2.h

```diff
@@ -91,8 +104,30 @@ struct preamble_block_t {
 static_assert(sizeof(preamble_block_t) % CRYPTO_BLOCK_SIZE == 0);
 static_assert(std::is_standard_layout<preamble_block_t>::value);
 
+// V2 epilogue conveys integrity/authentication information. It's added
+// at the end of each frame holds:
+//  * CRC32 for MAX_NUM_SEGMENTS -- in plain mode,
+//  * cipher-specific data (e.g. auth tag for AES-GCM).
+// Additionally, it consists flags field used currently to abort message
+// processing. It can be set by transceiver even very lately without any
+// need for introducing buffering.
+struct epilogue_block_t {
+  union {
+    char auth_tag[CRYPTO_BLOCK_SIZE];
```

Comment:
I spent sometime trying to find in the code where this `auth_tag` was being added to the payload bufferlist when encoding a frame, but couldn't find. Although when reading the epilogue from the wire and casting the bytes to this structure suddenly you use the `auth_tag`.
I'm guessing `auth_tag` is somekind of padding/signature bytes added by the encryption function and that's why I didn't saw it.

Having this in the epilogue feels misleading as any other person might feel a bit lost on how this encrypt/decrypt process is being done.
I would rather see a comment in the `SignedEncryptedFrame::Encode` saying that after encrypting the payload now has more `CRYPTO_BLOCK_SIZE` bytes, and in the read_frame code path see a different code path for reading the epilogue and the "remaining bytes" required for encrypted frames.

Reply:
Being terribly misleading is not the biggest flaw, to be honest. Initially the length of *signature* in secure mode (*auth tag* in the concrete, GCM-based implementation we have now) was an implementation detail of `ceph::crypto::onwire`. This provided flexibility for cipher selection. Somebody could exchange AES-GCM with e.g. AES-CBC + SHA512 and `ProtocolV2` wouldn't notice that. Gluing the 3 things (CRCs, *auth tag* and *late flags*) altogether erased this useful trait.